python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt_tests/test_op.py

```diff
@@ -0,0 +1,33 @@
+from dagster import OpExecutionContext, job, op
+from dagster_embedded_elt.sling import SlingReplicationParam
+from dagster_embedded_elt.sling.resources import SlingConnectionResource, SlingResource
+
+
+def test_base_sling_config_op(
+    csv_to_sqlite_replication_config: SlingReplicationParam,
+    path_to_temp_sqlite_db: str,
+):
+    sling_resource = SlingResource(
+        connections=[
+            SlingConnectionResource(type="file", name="SLING_FILE"),
+            SlingConnectionResource(
+                type="sqlite",
+                name="SLING_SQLITE",
+                connection_string=f"sqlite://{path_to_temp_sqlite_db}",
+            ),
+        ]
+    )
+
+    @op(out={})
+    def my_sling_op_yield_events(context: OpExecutionContext, sling: SlingResource):
+        yield from sling.replicate(
+            context=context, replication_config=csv_to_sqlite_replication_config
+        )
+
+    @job
+    def my_sling_op_yield_events_job():
+        my_sling_op_yield_events()
+
+    res = my_sling_op_yield_events_job.execute_in_process(resources={"sling": sling_resource})
+    assert res.success
+    assert len(res.get_job_success_event()) == 1
```

Comment:
@rexledesma I've added a test to use the Sling integration within an `op`. However, I'm not yielding `Output` from the `replicate()` method, is that alright?

When I included:
```
            if isinstance(context, OpExecutionContext):
                yield Output(
                    value=None, output_name=output_name.to_string(), metadata={"elapsed_time": end_time - start_time}
                )
```
I received the error:
```
E               dagster._core.errors.DagsterInvariantViolationError: Core compute for op "my_sling_op_yield_events" returned an output "["target", "main", "tbl"]" that does not exist. The available outputs are []
```
Which I supposed is expected when the decorator is `out={}`.

Reply:
That's fine. We shouldn't police folks on the exact outputs of their `@op`'s anyways. 

That should be for them to decide. If they want an `Output`, they should be fully in control of its definition and how its contents are passed downstream to another `@op`.

If they want to yield `Output`'s per each `AssetMaterialization`, that's doable. They just have to translate between those two classes themselves.

Here's how `dagster-dbt` does it:
https://github.com/dagster-io/dagster/blob/12a03cb890a39099ec6f61f9a789b483e0630a74/python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py#L206-L229